File: 002/002.go
package main

import (
    "fmt"
    "strconv"
)

func main() {

    fib := []int{1, 2}
    for fib[len(fib) - 1] < 4000000 {
        fib = append(fib, calc(fib))
    }

    var sum int
    for _, num := range fib {
        if num % 2 == 0 {
            sum += num
        }
    }

    fmt.Println(strconv.Itoa(sum))

}

func calc(s []int) int {
    return s[len(s) - 1] + s[len(s) - 2]
}

File: 002/002.rb
def calc(s)
    return s.at(s.length - 1) + s.at(s.length - 2)
end

fib = Array[1, 2]
while fib.at(fib.length - 1) < 4e6
    fib << calc(fib)
end

fib.delete_if { |n| n % 2 != 0 }
sum = fib.reduce(:+)
puts sum
